### ISSUE TITLE
PyImport_Import crash while using bazel to build the project

### ISSUE THREAD

**Issue by kanoncom** [tensorflow/tensorflow#11544]
I recently ran into a problem, when using bazel to build a project. when the project is compiled to application,no crash,everything works just fine,but,when it is compiled as a dynamic linking library,and called by another application ,core dump occurs every time.Is there anything wrong with my code or with the BUILD file ?
   here is the core dump information:
`55 0x00007f8189a4bd2c in PyImport_ExecCodeModuleEx () at /lib64/libpython2.7.so.1.0
#56 0x00007f8189a4bfa8 in load_source_module () at /lib64/libpython2.7.so.1.0
#57 0x00007f8189a4cc41 in import_submodule () at /lib64/libpython2.7.so.1.0
#58 0x00007f8189a4ce8d in load_next () at /lib64/libpython2.7.so.1.0
#59 0x00007f8189a4d86e in PyImport_ImportModuleLevel () at /lib64/libpython2.7.so.1.0
#60 0x00007f8189a30b1f in builtin___import__ () at /lib64/libpython2.7.so.1.0
#61 0x00007f81899a08e3 in PyObject_Call () at /lib64/libpython2.7.so.1.0
#62 0x00007f8189a326f7 in PyEval_CallObjectWithKeywords () at /lib64/libpython2.7.so.1.0
#63 0x00007f8189a373b5 in PyEval_EvalFrameEx () at /lib64/libpython2.7.so.1.0
#64 0x00007f8189a3be3d in PyEval_EvalCodeEx () at /lib64/libpython2.7.so.1.0
#65 0x00007f8189a3bf42 in PyEval_EvalCode () at /lib64/libpython2.7.so.1.0
#66 0x00007f8189a4bd2c in PyImport_ExecCodeModuleEx () at /lib64/libpython2.7.so.1.0
#67 0x00007f8189a4bfa8 in load_source_module () at /lib64/libpython2.7.so.1.0
#68 0x00007f8189a4cc41 in import_submodule () at /lib64/libpython2.7.so.1.0
#69 0x00007f8189a4ce8d in load_next () at /lib64/libpython2.7.so.1.0
#70 0x00007f8189a4d86e in PyImport_ImportModuleLevel () at /lib64/libpython2.7.so.1.0
#71 0x00007f8189a30b1f in builtin___import__ () at /lib64/libpython2.7.so.1.0
#72 0x00007f81899a08e3 in PyObject_Call () at /lib64/libpython2.7.so.1.0
---Type <return> to continue, or q <return> to quit---
#73 0x00007f81899a09c5 in call_function_tail () at /lib64/libpython2.7.so.1.0
#74 0x00007f81899a0aae in PyObject_CallFunction () at /lib64/libpython2.7.so.1.0
#75 0x00007f8189a4e312 in PyImport_Import () at /lib64/libpython2.7.so.1.0
#76 0x00007f818b8ef7fc in load_main () at ./test.so
#77 0x0000000000400826 in main() () at test1.cpp:7`

here is the console information:
`F tensorflow/core/framework/function.cc:1015] Check failed: GetOpGradFactory()->insert({op, func}).second Duplicated gradient for Softmax
Aborted (core dumped)`

here is my code:
`    Py_Initialize();
    if ( !Py_IsInitialized() ) {
        return -1;
    }
    PyEval_InitThreads();
    PyThreadState *mainThreadState = NULL;
    //  save a pointer to the main PyThreadState object 
    mainThreadState = PyThreadState_Get();
    //  release the lock 
    PyEval_ReleaseLock();
    char* mockargv[1]={(char*)""};
    PySys_SetArgv(1,mockargv);
    PyRun_SimpleString("import sys");
    PyRun_SimpleString("sys.path.append('./')");
    pName_ = PyString_FromString("run");
    displayPyObject(pName_);
    if(pName_ == NULL){
        return -1;
    }
    pModule_ = PyImport_Import(pName_);
    displayPyObject(pModule_);`

here is my BUILD file
`cc_binary(
    name = "test2",
    visibility = ["//visibility:__subpackages__"],
srcs = glob(["test.cpp"
             ],
        ),
        includes=["test.h",
                "/usr/include/python2.7/",
],
   linkopts = [
"-O2 -lm -lpthread -L/usr/lib/python2.7 -lpython2.7 -lopencv_core -lopencv_imgproc -lopencv_highgui -lopencv_ml -lfreeimage",
],
deps = [
        "//tensorflow/cc:cc_ops",
        "//tensorflow/core:framework",
        "//tensorflow/core:framework_internal",
        "//tensorflow/core:tensorflow",
    ],
)
`
and here is the dynamic linking library part of my BUILD file
`cc_binary(
    name = "test.so",
    linkshared = 1,
    deps = [
        ":test_lib",
    ],
)
cc_library(
    name = "test_lib",
    visibility = ["//visibility:__subpackages__"],
    srcs = glob(["test.cpp"
             ],
        ),
    includes=["test.h"],
    linkopts = [
"-lm -lpthread -L/usr/lib/python2.7 -lpython2.7 -lopencv_core -lopencv_imgproc -lopencv_highgui -lopencv_ml -lfreeimage"
    ],
    deps = [
        "//tensorflow/cc:cc_ops",
        "//tensorflow/core:framework",
        "//tensorflow/core:framework_internal",
        "//tensorflow/core:tensorflow",
    ],
)
`

**Comment by asimshankar** [comment]
I apologize but I am having a hard time understanding what the problem is, where the problem is, and what version it affects. Please resubmit and pay attention to the issue template (https://github.com/tensorflow/tensorflow/issues/new) . Please provide all the information it asks. Thank you.
